pkg/sql/colexec/external/external.go

```diff
@@ -209,6 +215,127 @@ func InitS3Param(param *tree.ExternParam) error {
 	return nil
 }
 
+func containColname(col string) bool {
+	return strings.Contains(col, STATEMENT_ACCOUNT) || strings.Contains(col, catalog.ExternalFilePath)
+}
+
+func judgeContainColname(expr *plan.Expr) bool {
+	expr_F, ok := expr.Expr.(*plan.Expr_F)
```

Comment:
why do you need to special handle expression?    Can it be that if the expr is a const, that can be evalued on const string, then we call the expression evaluation to eval it.    for example, I can do a substring on __mo_filepath, or I can even evaluate regular expression.

Reply:
Why "or" need to handled separately?   Does a generic expr evaluation on const value work?

Reply:
Or say it in a different way, this code should be using the const folding code -- after supplying filepath value, if the expr can be folded to constant false, we can skip the file.

Reply:
are you able to do __mo_filepath like '%foo%' and __mo_filepath ilike '%bar%'?

Reply:
and for __mo_filepath like '%a.txt', you should not parse b.txt.    Can we test this?

For example, let's have a invalid file error.txt, if it parse that file it will fail.   By setting __mo_filepath like '%a.txt' we should run the query.